public/javascripts/registration.js

```diff
@@ -1,18 +1,62 @@
+'use strict';
 $(document).ready(function () {
+    var minDateTick = Date.parse('2020-01-10');
+
     var model = {
         name: ko.observable(),
         nameTouched: ko.observable(false),
+        date: ko.observable(),
+        dateTouched: ko.observable(false),
         comment: ko.observable(),
         photoUrl: ko.observable(),
-        showPhotoRequired: ko.observable(false),
+        showPhotoRequired: ko.observable(false)
     };
-    model.onNameChange = function (data, event) {
+    model.onNameChange = function () {
         model.nameTouched(true);
     };
+    model.isNameValid = function () {
+        return model.name().length > 0;
+    };
     model.showNameRequired = ko.pureComputed(function () {
-        return model.nameTouched() && model.name().length === 0;
-    }),
-    model.onPhotoChange = function (data, event) {
+        return model.nameTouched() && !model.isNameValid();
+    });
+    model.onDateChange = function () {
+        model.dateTouched(true);
```

Comment:
the date is not pre-filled so there is no change on submit and the (server-side) error is displayed first after the upload